### PR TITLE
feat: switch disableTelemetry to bunfig

### PR DESCRIPTION
**Pull request by coratgerl**
### What does this PR do?
Closes: https://github.com/oven-sh/bun/issues/5683

This PR switch ```disableTelemetry```to bunfig instead of .env.

- [x] Documentation or TypeScript types (it's okay to leave the rest blank in this case)
- [x] Code changes

### How did you verify your code works?
Manual tests

- [x] I checked the lifetime of memory allocated to verify it's (1) freed and (2) only freed when it should be
- [x] I or my editor ran `zig fmt` on the changed files

### CODE REVIEW

**Comment by liz3** [comment]
Maybe keeping the option to set it via a environment variable would be a better approach, so that both can be used, since while in a config it makes the most logical sense, having the ability to set it via a environment variable as a second option is a more "complete" approach.

Maybe there will be scenarios where it's supposed to be disabled in a configuration but a certain direct execution(shell) wants it enabled, arguably thats a edge case but the possibility does exist.

File: docs/runtime/env.md [hunk 1]
```diff
@@ -75,10 +75,3 @@ These environment variables are read by Bun and configure aspects of its behavio
 
 - `FORCE_COLOR`
 - If `FORCE_COLOR=1`, then ANSI color output is force enabled, even if `NO_COLOR` is set.
-
----
-
-- `DO_NOT_TRACK`
```
Comment (Electroid) [inline docs/runtime/env.md:81]: We should keep environment variable.

File: src/bunfig.zig [hunk 2]
```diff
@@ -207,6 +207,11 @@ pub const Bunfig = struct {
                 if (json.get("preload")) |expr| {
                     try this.loadPreload(allocator, expr);
                 }
+
+                if (json.get("disableTelemetry")) |expr| {
```
Comment (Electroid) [inline src/bunfig.zig]: Instead of inverting the logic, we can just do `telemetry = false`

File: docs/runtime/env.md (hunk 1 above)
Comment (coratgerl) [reply docs/runtime/env.md:81]: Ok thank you, I will revert this :)

File: src/bunfig.zig (hunk 2 above)
Comment (coratgerl) [reply src/bunfig.zig]: Thank you for your feedback, I will fix this :)

**Review by Electroid — APPROVED** [review]
Thanks!

**Pull request merged**